compiler/rustc_interface/src/passes.rs

```diff
@@ -110,12 +110,9 @@ mod boxed_resolver {
     }
 
     impl BoxedResolver {
-        pub(super) fn new<F>(session: Lrc<Session>, make_resolver: F) -> Result<(ast::Crate, Self)>
+        pub(super) fn new<F>(session: Lrc<Session>, make_resolver: F) -> Self
```

Comment:
```suggestion
        pub(super) fn new(session: Lrc<Session>, make_resolver: impl for<'a> FnOnce(&'a Session, &'a ResolverArenas<'a>) -> Resolver<'a>) -> BoxedResolver
```
`impl Trait` can be used here + `Self` always makes code less readable unless the code is generic or the type is really unwieldy.